### DISCUSSION TITLE
Allow flexibility and more examples for Audio Stream

### DISCUSSION THREAD

**Discussion by jmorel33** [raysan5/raylib#1881] [Ideas]
I've been on and off studying RayLib and the place I struggle the most is with the audio.  It is so incomplete and hard to work with.

Right now I just want to funnel in my own audio stream.

The stream example is hard to understand and clunky, hard to adapt to any situation.

How do we get the bitrate and depth of the audio device ?  This would allow passing those values to initialize the stream.
(something similar as what is done for the Graphics device for resolution bit depth)

![image](https://user-images.githubusercontent.com/8346799/126054030-10c6fedc-018a-4f27-b3ee-ec45438f908e.png)

I am a musician and this is my passion.

Having so many stop blocks and minimal audio functionalities makes me want to create them but I have nowhere to go with what is supplied.

Anny help is welcomed.

Thank.

**Comment by chriscamacho** [comment]
raylib was never designed to be a [DAW](https://en.wikipedia.org/wiki/Digital_audio_workstation) 

the audio system is primarily designed to play background music and game sound effects, before I contributed the multi channel audio pool, individual sound effects weren't even multitimbral

If you are wanting to create an application for serious audio work, you can always use another library (that might support VST's for example) while leaving raylib's audio subsystem uninitialised

**Reply by jmorel33** [reply]
> raylib was never designed to be a [DAW](https://en.wikipedia.org/wiki/Digital_audio_workstation)
> 
> the audio system is primarily designed to play background music and game sound effects, before I contributed the multi channel audio pool, individual sound effects weren't even multitimbral
> 
> If you are wanting to create an application for serious audio work, you can always use another library (that might support VST's for example) while leaving raylib's audio subsystem uninitialised

@chriscamacho
You could have just written: go look elsewhere, that way  you would  waste less time as clearly you have not read my message.

#1 I explained that the "The stream example is hard to understand and clunky, hard to adapt to any situation."

#2 I asked if there is a way to "get the bitrate of the audio device"

#3 I mentioned "I am a musician and this is my passion.".

#4 Where di you read I asked for a complete overhaul of the music system?  Or that I want it to act like a DAW?  I have a DAW I don't need RayLib to be my DAW.

If I wanted new features I would list them here:
- Stereo panning control
- Reverb, Delay, Chorus, Doppler, 3D spacialization etc...
- etc.

If sound and music is not your passion, I understand.

Thank you.

**Reply by chriscamacho** [reply]
audio is *a* passion of mine, please *read* the second paragraph of my reply (which you obviously didn't)

when I implemented multichannel audio in raylib some time ago, I looked at the feasibility of not only getting some extra info out of the audio device and maybe adding some simple effects too, however I decided that was *way* outside the aims (as I understand them) of raylib (please read the second paragraph of my initial reply)

judging from your reply I see ignorance is also a passion of yours

**Reply by jmorel33** [reply]
> audio is _a_ passion of mine, please _read_ the second paragraph of my reply (which you obviously didn't)
> 
> when I implemented multichannel audio in raylib some time ago, I looked at the feasibility of not only getting some extra info out of the audio device and maybe adding some simple effects too, however I decided that was _way_ outside the aims (as I understand them) of raylib (please read the second paragraph of my initial reply)
> 
> judging from your reply I see ignorance is also a passion of yours

DID YOU READ MY 1ST COMMENT?
> You could have just written: go look elsewhere, that way you would waste less time as clearly you have not read my message.

And basically you just reiterated the same message + insults.

You won't get my sympathy with your way of acting.

If you think your implementation was flawless, think again.  I had to dig deep to get a few major flaws fixed, thankfully that happened otherwise I would have never used it.

On July 10 I asked you a question on this other thread, and never got feedback>
https://github.com/raysan5/raylib/discussions/1663

Essentially,  your way or the highway.

Cheers.
